includes/class-amp-service-worker.php

```diff
@@ -0,0 +1,316 @@
+<?php
+/**
+ * AMP Service Workers.
+ *
+ * @package AMP
+ * @since 1.1
+ */
+
+/**
+ * Class AMP_Service_Worker.
+ */
+class AMP_Service_Worker {
+
+	/**
+	 * Query var that is used to signal a request to install the service worker in an iframe.
+	 *
+	 * @link https://www.ampproject.org/docs/reference/components/amp-install-serviceworker#data-iframe-src-(optional)
+	 */
+	const INSTALL_SERVICE_WORKER_IFRAME_QUERY_VAR = 'amp_install_service_worker_iframe';
+
+	/**
+	 * Init.
+	 */
+	public static function init() {
+		if ( ! class_exists( 'WP_Service_Workers' ) ) {
+			return;
+		}
+
+		$theme_support = AMP_Theme_Support::get_theme_support_args();
+		if ( empty( $theme_support['service_worker'] ) ) {
```

Comment:
I wonder if we should opt-in or opt-out? If they have the PWA plugin, perhaps this should be enough of a signal that they want this functionality? In that case, this could be:

```suggestion
		if ( isset( $theme_support['service_worker'] ) && false === $theme_support['service_worker'] ) {
```